lib/enmeshed/connector_api.rb

```diff
@@ -0,0 +1,244 @@
+# frozen_string_literal: true
+
+module Enmeshed
```

Comment:
I was considering two additional changes to this module:

1. We could extract the `Relationship` class (and the `RelationshipTemplate` if added) to a dedicated file. I'd say that they have become large enough to justify that.
2. We could rename the `ConnectorApi` to `Connector`, since the class is our interface to the connector and the only mechanism we communicate with it.